test/framework/debug.go

```diff
@@ -291,64 +301,97 @@ func configDump(kumactlOpts kumactl.KumactlOptions, debugPath string, cluster Cl
 	}
 	if err != nil {
 		dpInspectError = fmt.Sprintf("kumactl get dataplanes failed with error: %s", err.Error())
-		errorSeen = true
 	} else {
 		if jsonErr := json.Unmarshal([]byte(dpListJson), &dpResp); jsonErr != nil {
 			dpInspectError = fmt.Sprintf("json Unmarshal dataplane list failed with error: %s", jsonErr.Error())
-			errorSeen = true
-		} else {
-			for _, dpObj := range dpResp.Items {
-				var dpNS string
-				dpNameParts := strings.Split(dpObj.Name, ".")
-				if len(dpNameParts) > 1 {
-					dpNS = dpNameParts[1]
-				}
-				if dpNS == "" {
-					continue
-				}
-				if !namespaceExported(debugPath, cluster.Name(), dpNS) {
-					continue
-				}
+		}
+	}
 
-				configDumpResp := ""
```

Comment:
Code here is extracted into the new function `doInspect` underneath. In that function, I implementd the ability to inspect different types info.